bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/firmware/FirmwareImpl.java

```diff
@@ -139,6 +146,11 @@ public String getPrerequisiteVersion() {
         return (prerequisiteVersion != null) ? prerequisiteVersion.toString() : null;
     }
 
+    @Override
+    public FirmwareRestriction getCustomRestrictions() {
```

Comment:
The plural `s` suggests a collection is returned. Just like the class `FirmwareRestriction` the getter (and the field)  should also be named `firmwareRestriction` to indicate a single instance is involved.